src/metabase/middleware.clj

```diff
@@ -60,14 +60,18 @@
   (fn [{:keys [metabase-session-id] :as request}]
     ;; TODO - what kind of validations can we do on the sessionid to make sure it's safe to handle?  str?  alphanumeric?
     (handler (or (when (and metabase-session-id ((resolve 'metabase.core/initialized?)))
-                   (when-let [session (db/select-one [Session :created_at :user_id]
+                   (when-let [session (db/select-one [Session :created_at :user_id (db/qualify User :is_superuser)]
                                         (db/join [Session :user_id] [User :id])
                                         (db/qualify User :is_active) true
                                         (db/qualify Session :id) metabase-session-id)]
-                     (let [session-age-ms (- (System/currentTimeMillis) (.getTime ^java.util.Date (get session :created_at (u/->Date 0))))]
+                     (let [session-age-ms (- (System/currentTimeMillis) (or (when-let [^java.util.Date created-at (:created_at session)]
+                                                                              (.getTime created-at))
+                                                                            0))]
                        ;; If the session exists and is not expired (max-session-age > session-age) then validation is good
                        (when (and session (> (config/config-int :max-session-age) (quot session-age-ms 60000)))
-                         (assoc request :metabase-user-id (:user_id session))))))
+                         (assoc request
+                           :metabase-user-id (:user_id session)
+                           :is-superuser?    (:is_superuser session))))))
```

Comment:
Does this mean that if a user has a session running and their account is stripped of the superuser bit, we'd need to wait `max-session-age` minutes until they can't fuck things up?